src/main/java/duke/command/Ui.java

```diff
@@ -0,0 +1,124 @@
+package duke.command;
+
+import java.util.Scanner;
+
+import duke.exception.InvalidCommandException;
+/**
+ * This class handles the interactions with the user.
+ */
+public class Ui {
+    private static final String LINE_HORIZONTAL =
+            "___________________________________________________________";
+    private static Scanner sc;
+    private String botName;
+    private Parser parser;
+
+    /**
+     * Constructor to create a new UI.
+     *
+     * @param name The name of the bot.
+     */
+    public Ui(String name) {
+        this.botName = name;
+        sc = new Scanner(System.in);
+        this.parser = new Parser();
+    }
+
+    /**
+     * Prints initial message as prompt.
+     */
+    public void printWelcomeMessage() {
+        System.out.printf("Greetings! This is %s.\n"
+                + "What can I do for you?\n",
+                this.botName);
+    }
+
+    /**
+     * Prints loading error message.
+     */
+    public void printLoadingError() {
+        System.out.printf("%s\nIt seems there was an error reading the saved list.\n"
+                + "Please ensure a duke.txt file is present in /data.\n"
+                + "%s\n",
+                LINE_HORIZONTAL, LINE_HORIZONTAL);
+    }
+
+    /**
+     * Takes in user input and returns it.
+     *
+     * @return The trimmed input.
+     */
+    public String getInput() {
+        return sc.nextLine().trim();
+    }
+
+    /**
+     * Takes in user input and performs an action accordingly.
+     *
+     * @param input The user's input.
+     * @throws InvalidCommandException If the command is unrecognised.
+     */
+    public String receiveUserCommand(String input) throws InvalidCommandException {
+        return this.parser.interpretCommand(input);
+    }
+
+    /**
+     * Prints the message for various types of exceptions.
+     *
+     * @param type The type of exception.
+     */
+    public void printException(String type) {
```

Comment:
I like how you use the switch statements here. I can look into implementing this!